### PR TITLE
feat(memory): implement LifecycleManager and CleanupManager (task-384, task-393)

### PR DESCRIPTION
**Pull request by jpoley**
## Summary
- Implements LifecycleManager for task memory lifecycle management (task-384)
- Implements CleanupManager for automated memory archiving and deletion (task-393)
- Adds path traversal security protection (CWE-22) to TaskMemoryStore.get_path()

### LifecycleManager (`src/specify_cli/memory/lifecycle.py`)
- Orchestrates memory operations on task state transitions
- Creates memory when task moves To Do → In Progress
- Archives memory when task completes (In Progress → Done)
- Deletes memory when task is archived (Done → Archive)
- Restores memory when task is reopened (Done → In Progress)
- Deletes memory when task is reset (In Progress → To Do)
- Updates backlog/CLAUDE.md with @import for active task context

### CleanupManager (`src/specify_cli/memory/cleanup.py`)
- Archive active memories older than N days
- Delete archived memories older than N days
- Dry-run mode for safe testing
- Statistics: active/archived counts, oldest ages
- cleanup_all() convenience method with configurable thresholds

### Security Fix
- Add regex validation to TaskMemoryStore.get_path() to prevent path traversal attacks
- Validate task_id matches `task-NNN` pattern (numeric only)

## Test Plan
- [x] All 109 memory tests pass
- [x] Path traversal protection tests added
- [x] LifecycleManager state transition tests (25 tests)
- [x] CleanupManager age-based cleanup tests (21 tests)
- [x] Edge cases and error handling covered

## Related Tasks
- task-384: Implement LifecycleManager Component
- task-393: Implement CleanupManager Component
- Supersedes failed PR from host-task-integration branch

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/specify_cli/memory/cleanup.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        active_count = len(active_tasks)
+
+        oldest_active_days = 0
+        if active_tasks:
+            oldest_active_mtime = min(
+                datetime.fromtimestamp(self.store.get_path(task_id).stat().st_mtime)
+                for task_id in active_tasks
+                if self.store.get_path(task_id).exists()
+            )
+            oldest_active_days = (now - oldest_active_mtime).days
```
[the remark this answers is not in the document]
Comment (jpoley) [reply src/specify_cli/memory/cleanup.py]: fix me

**Comment by jpoley** [comment]
fixes needed
